src/iceberg/table_metadata.cc

```diff
@@ -153,14 +155,42 @@ Result<MetadataFileCodecType> TableMetadataUtil::CodecFromFileName(
   return MetadataFileCodecType::kNone;
 }
 
+Result<std::string> DecompressGZIPFile(const std::string& filepath) {
+  gzFile file = gzopen(filepath.c_str(), "rb");
+  if (!file) {
+    return IOError("Failed to open gzip file:{} ", filepath);
+  }
+
+  static const int CHUNK_SIZE = 32768;  // 32KB chunks
+  std::array<char, CHUNK_SIZE> buffer;
+  std::string result;
+  int bytes_read;
+
+  while ((bytes_read = gzread(file, buffer.data(), CHUNK_SIZE)) > 0) {
+    result.append(buffer.data(), bytes_read);
+  }
+
+  int err;
+  const char* error_msg = gzerror(file, &err);
+  if (err != Z_OK) {
+    gzclose(file);
+    return IOError("Error during gzip decompression:{} ", std::string(error_msg));
```

Comment:
I'm not convinced that `error_msg` is valid after `file` is closed.

Reply:
In fact, if you peek at the implementation it points into the file structure

https://github.com/madler/zlib/blob/5a82f71ed1dfc0bec044d9702463dbdf84ea3b71/gzlib.c#L489-L505